homeassistant/components/epson/__init__.py

```diff
@@ -3,40 +3,53 @@
 import logging
 
 from epson_projector import Projector
-from epson_projector.const import POWER, STATE_UNAVAILABLE as EPSON_STATE_UNAVAILABLE
+from epson_projector.const import (
+    PWR_OFF_STATE,
+    STATE_UNAVAILABLE as EPSON_STATE_UNAVAILABLE,
+)
 
 from homeassistant.components.media_player import DOMAIN as MEDIA_PLAYER_PLATFORM
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.const import CONF_HOST
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
-from .const import DOMAIN
-from .exceptions import CannotConnect
+from .const import DOMAIN, HTTP
+from .exceptions import CannotConnect, PoweredOff
 
 PLATFORMS = [MEDIA_PLAYER_PLATFORM]
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def validate_projector(hass: HomeAssistant, host, port):
-    """Validate the given host and port allows us to connect."""
+async def validate_projector(
+    hass: HomeAssistant, host, check_power=True, check_powered_on=True
+):
+    """Validate the given projector host allows us to connect."""
     epson_proj = Projector(
         host=host,
         websession=async_get_clientsession(hass, verify_ssl=False),
-        port=port,
+        type=HTTP,
     )
-    _power = await epson_proj.get_property(POWER)
-    if not _power or _power == EPSON_STATE_UNAVAILABLE:
-        raise CannotConnect
+    if check_power:
+        _power = await epson_proj.get_power()
+        if not _power or _power == EPSON_STATE_UNAVAILABLE:
+            _LOGGER.debug("Cannot connect to projector")
+            raise CannotConnect
+        if _power == PWR_OFF_STATE and check_powered_on:
+            _LOGGER.debug("Projector is off")
+            raise PoweredOff
     return epson_proj
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
```

Comment:
We should also set unique id of existing config entries that don't have that set yet.

Reply:
It will be done by media_player entity, right?